src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -210,6 +210,23 @@
     static AcceptVersions accept_messaging = new AcceptVersions(minimum_version, current_version);
     static AcceptVersions accept_streaming = new AcceptVersions(current_version, current_version);
 
+    public static int[] supportedVersions()
+    {
+        return new int[] { MessagingService.VERSION_30, MessagingService.VERSION_3014,
+                           MessagingService.VERSION_40};
+    }
+
+    public static String versionToHumanReadable(int version)
+    {
+        switch (version)
+        {
+            case VERSION_30: return "VERSION_30";
+            case VERSION_3014: return "VERSION_3014";
+            case VERSION_40: return "VERSION_40";
+            default: throw new IllegalArgumentException("Unknown version: " + version);
```

Comment:
nit: I noticed that `versionToHumanReadable()` is only used in conjunction w/ `supportedVersions()`. It might make sense to formalize an inner `MessagingService.Version` enum here. I guess we'd at least avoid having to worry about the switch statement. (The version numbers already start at 10, so we won't be tempted to use the enum ordinals anywhere.)

Reply:
thought about that, but didn't want to update any usage... can try to see how invasive that would be

Reply:
I have mixed feelings about this.  I tested out where the enum held the value and updated the int constants to reuse, but this caused issues with switch statements (constants required).  I checked usage and the version is all over the place, so switching to enum would be a large patch; this then left me with a patch where I add enums but they copy the values and only replace these two methods... for some reason this feels dirty though... made the change though.